revised test-fn_dy test

sbohora opened the pull request.

camorosi commented on tests/testthat/test-dn_dry.R:
```diff
@@ -0,0 +1,52 @@
+    fn_dry(
+      creds_set_labkey(keyring::key_get(service = "API_KEY", keyring="remote"))
+    ),
+    "creds_set_labkey(keyring::key_get(service = \"API_KEY\", keyring = \"remote\"))"
```
Recommend switching to `service = \"LABKEY_BOARD\"` since the service would be referencing the board itself, not the key

camorosi commented on tests/testthat/test-dn_dry.R:
```diff
@@ -0,0 +1,52 @@
+testthat::test_that("This would test if aws credentials from keyring pass in fn_dry function",{
+  testthat::expect_equal(
+    fn_dry(
+      creds_set_aws(keyring::key_get(service = "AWS_KEY", keyring="remote"))
```
This is a valid check of fn_dry but an incorrect use of creds_set_aws because it only provides the key, not the secret.

sbohora replied:
Done. https://github.com/amashadihossein/dpbuild/commit/8e37f9dc671b6d6a971db5be09feb4ea4205baea

camorosi commented on R/dp_init.R:
```diff
@@ -205,11 +224,27 @@ dpconf_init <- function(project_path,
+      }
+    }
+    dried_fn <- rlang::expr_deparse(fn_as_call, width = 300)
+  } else {
```
Code streamlining suggestion: could remove the else statement and have a single call to expr_deparse outside of the if statement to avoid repeated code.

sbohora replied:
Done. That was a great advice. I ended of revising the fn_dry again to restrict some possible loopholes.

https://github.com/amashadihossein/dpbuild/commit/8e37f9dc671b6d6a971db5be09feb4ea4205baea

camorosi commented on R/dp_init.R:
```diff
@@ -205,11 +224,27 @@ dpconf_init <- function(project_path,
+
+    for (arg in get_fn_args) {
+      if (!rlang::is_callable(arg)) {
+        stop(cli::format_error(glue::glue("Do not supply the credentials directly as the function arguments. Function arguments in {fn_as_call_main} need to be passed as function calls. For example, Sys.getenv()")))
```
It might be helpful to be more specific in how to pass credentials via Sys.getenv()

Suggest something more like `key = Sys.getenv(\"API_KEY\")` instead of simply `Sys.getenv()`

sbohora replied:
Done.

camorosi commented on R/dp_init.R:
```diff
@@ -77,6 +77,25 @@ dp_init <- function(project_path = fs::path_wd(),
   wd0 <- fs::path_wd()
 
+  if (!rlang::is_character(x = creds_set_dried)){
+    message("Encountered error in creds_set_dried in dp_init")
+    message("Make sure to use fn_dry in argument passed to creds_set_dried paramater.")
+    stop(cli::format_error(glue::glue("Do not supply the credentials directly as the function arguments.")))
+  }
+
+  if (!rlang::is_character(x = board_params_set_dried)){
+    message("Encountered error in board_params_set_dried in dp_init")
+    stop(cli::format_error(glue::glue("Make sure to use fn_dry in argument passed to board_params_set_dried paramater.")))
+  }
+
+  creds_set_dried_parsed <- rlang::parse_expr(creds_set_dried)
+
+  if (!class(creds_set_dried_parsed) == "call") {
+    message("Encountered error in creds_set_dried in dp_init")
+    message("Make sure to use fn_dry in argument passed to creds_set_dried paramater.")
+    stop(cli::format_error(glue::glue("Do not supply the credentials directly as the function arguments.")))
```
For each if statement, recommend combining the message and stop calls into a single stop call for each, using glue for multi-line error messages.

sbohora replied:
Done! Thanks.

camorosi reviewed (commented):
This all looks great! I added a few small in-line recommendations for streamlining and clarity. 

Also, small typo in "tests/testthat/test-dn_dry.R", should be "tests/testthat/test-fn_dry.R"

sbohora commented:
> This all looks great! I added a few small in-line recommendations for streamlining and clarity.
> 
> Also, small typo in "tests/testthat/test-dn_dry.R", should be "tests/testthat/test-fn_dry.R"

Corrected. Thanks @camorosi

amashadihossein reviewed (approved):
This looks great!
